### PR TITLE
Symfony 7.0 support and refactoring

### PR DESCRIPTION
**Pull request by michnovka**
Strong typed everything. Symfony 7 support. Dropped Symfony 5 support. Bump min PHP version to 8.0.2. Minor logic fixes. Migrate to new ECS version.

### CODE REVIEW

File: Generator/ImageFileHandler.php [hunk 1]
```diff
@@ -74,7 +62,7 @@ public function saveAsFile($contents): string
 
     public function collectGarbage(): bool
     {
-        if (1 == !mt_rand(1, $this->gcFreq)) {
+        if (1 != mt_rand(1, $this->gcFreq)) {
```
Comment (michnovka) [inline Generator/ImageFileHandler.php:65]: @Gregwar I think this had to be a long-time bug in the code, as the old code would be ALWAYS false. Hence the garbage collection was running EVERY SINGLE TIME. But check, maybe I didnt understant properly

File: ecs.php
Comment (michnovka) [inline ecs.php:1]: I was unable to port completely all ECS rules from the old YAML file (dropped support for YAML was with ECS 9). But I did copy most of them. Some are even discontinued now. The ECS <=7 did not support PHP 8, so I had to update it

File: composer.json [hunk 2]
```diff
@@ -9,21 +9,21 @@
...
+            "homepage": "https://www.gregwar.com/"
         },
         {
             "name": "Jeremy Livingston",
             "email": "jeremy.j.livingston@gmail.com"
         }
     ],
     "require": {
-        "php": ">=7.1.3",
+        "php": ">=8.0.2",
```
Comment (michnovka) [inline composer.json:20]: This is the PHP version required by Symfony 6.0

**Review by michnovka — COMMENTED** [review]
Please review my explanation for some of the controversial changes

File: composer.json [hunk 3]
```diff
@@ -9,21 +9,21 @@
...
         "ext-gd": "*",
-        "gregwar/captcha": "^1.1.9",
-        "symfony/form": "~5.0|~6.0",
-        "symfony/framework-bundle": "~5.0|~6.0",
-        "symfony/translation": "~5.0|^6.0",
-        "twig/twig": "^2.10|^3.0"
```
[the remark this answers is not in the document]
Comment (YoannChabert) [reply composer.json:26]: I would even force Symfony 6.0 to 6.4, as it will be the only supported in ~6

**Comment by michnovka** [comment]
@Gregwar can you please review this? We are stuck with symfony 6.x just because of this package. thanks!

File: Generator/ImageFileHandler.php (hunk 1 above)
Comment (ajgarlag) [reply Generator/ImageFileHandler.php:65]: I think you should use strict comparison here
```suggestion
        if (1 !== mt_rand(1, $this->gcFreq)) {
```

**Comment by Gregwar** [comment]
Sounds good to me, thanks for contributing
I tagged `v2.2.1`

@michnovka can you confirm me everything it's ok ?

**Comment by ajgarlag** [comment]
@Gregwar  The `v2.2.1` tag conflicts with the README.md compatibility table. The PR expects the new version to be tagged as `v2.3.0`

**Comment by Gregwar** [comment]
@ajgarlag I deleted `v2.2.1` and re-tagged `v2.3.0`

**Pull request merged**